File: internal/dto/approval.go
package dto

// CreateApprovalRequest 创建审批工单请求
type CreateApprovalRequest struct {
	Title       string `json:"title" binding:"required"`
	Description string `json:"description"`
	Type        string `json:"type" binding:"required"` // batch_delete / ssh_key_revoke / departure_cleanup / ssh_key_rotate
	Payload     string `json:"payload" binding:"required"`
}

// ReviewApprovalRequest 审批请求
type ReviewApprovalRequest struct {
	Action string `json:"action" binding:"required"` // approve / reject
	Note   string `json:"note"`
}

// ApprovalResponse 审批工单响应
type ApprovalResponse struct {
	ID            uint   `json:"id"`
	Title         string `json:"title"`
	Description   string `json:"description"`
	Type          string `json:"type"`
	Payload       string `json:"payload"`
	Status        string `json:"status"`
	RequestedBy   uint   `json:"requested_by"`
	RequesterName string `json:"requester_name"`
	ReviewedBy    *uint  `json:"reviewed_by"`
	ReviewerName  string `json:"reviewer_name"`
	ReviewedAt    string `json:"reviewed_at"`
	ReviewNote    string `json:"review_note"`
	CreatedAt     string `json:"created_at"`
}

File: web/src/services/approval.ts
import request from '../utils/request'
import type { PageResult } from './audit'

export interface ApprovalItem {
  id: number
  title: string
  description: string
  type: string
  payload: string
  status: string
  requested_by: number
  requester_name: string
  reviewed_by: number | null
  reviewer_name: string
  reviewed_at: string
  review_note: string
  created_at: string
}

export interface CreateApprovalParams {
  title: string
  description?: string
  type: string
  payload: string
}

export interface ReviewApprovalParams {
  action: 'approve' | 'reject'
  note?: string
}

export function getApprovals(params?: {
  status?: string
  type?: string
  page?: number
  page_size?: number
}): Promise<PageResult<ApprovalItem>> {
  return request.get('/approvals', { params })
}

export function getMyApprovals(params?: {
  page?: number
  page_size?: number
}): Promise<PageResult<ApprovalItem>> {
  return request.get('/approvals/mine', { params })
}

export function getApproval(id: number): Promise<ApprovalItem> {
  return request.get(`/approvals/${id}`)
}

export function createApproval(data: CreateApprovalParams): Promise<ApprovalItem> {
  return request.post('/approvals', data)
}

export function reviewApproval(id: number, data: ReviewApprovalParams): Promise<ApprovalItem> {
  return request.put(`/approvals/${id}/review`, data)
}

export function cancelApproval(id: number): Promise<void> {
  return request.put(`/approvals/${id}/cancel`)
}

export function getPendingCount(): Promise<{ count: number }> {
  return request.get('/approvals/pending-count')
}
